modmesh/app/naca.py

```diff
@@ -0,0 +1,144 @@
+# Copyright (c) 2021, Yung-Yu Chen <yyc@solvcon.net>
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# - Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+# - Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+# - Neither the name of the copyright holder nor the names of its contributors
+#   may be used to endorse or promote products derived from this software
+#   without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+
+"""
+Show NACA airfoil shape
+"""
+
+import numpy as np
+
+from .. import core
+from .. import view
+from .. import apputil
+
+
+def calc_naca4_points(number, npoint, open_trailing_edge=False,
+                      cosine_spacing=False):
+    """
+    Returns points in [0 1] for the given 4 digit NACA number string.
+    Reference from
+    https://ntrs.nasa.gov/api/citations/19930091108/downloads/19930091108.pdf
+    and https://github.com/dgorissen/naca
+    """
+
+    camber = float(number[0]) / 100.0
+    pos = float(number[1]) / 10.0
+    thick = float(number[2:]) / 100.0
+
+    a0 = +0.2969
+    a1 = -0.1260
+    a2 = -0.3516
+    a3 = +0.2843
+    a4 = -0.1015 if open_trailing_edge else -0.1036
+
+    if cosine_spacing:
+        xc = 0.5 * (1.0 - np.cos(np.linspace(0.0, np.pi, npoint + 1)))
+    else:
+        xc = np.linspace(0.0, 1.0, npoint + 1)
+
+    xh = np.sqrt(xc)
+    x2 = np.power(xc, 2)
+    x3 = np.power(xc, 3)
+    x4 = np.power(xc, 4)
+    yt = 5.0 * thick * (a0 * xh + a1 * xc + a2 * x2 + a3 * x3 + a4 * x4)
```

Comment:
This is the formula described in https://ntrs.nasa.gov/api/citations/19930091108/downloads/19930091108.pdf .  Later I would like to fit it using Bezier curves.